File: moderate/stack.lua
function push(stack, int)
  stack[#stack+1] = int
  return
end

function pop(stack)
  local int = stack[#stack]
  stack[#stack] = nil
  return int
end

for line in io.lines(arg[1]) do
  local stack = {}
  for i in line:gmatch("%S+") do
    push(stack, tonumber(i))
  end
  local first = true
  while #stack > 0 do
    if first then first = false else io.write(" ") end
    io.write(pop(stack))
    pop(stack)
  end
  print()
end

File: moderate/stack.rb
def push(stack, int)
  stack << int
end

def pop(stack)
  int = stack[-1]
  stack.delete_at(-1)
  int
end

File.open(ARGV[0]).each_line do |line|
  stack = []
  line.split.map(&:to_i).each { |i| push(stack, i) }

  out = []
  while stack.length > 0
    out << pop(stack)
    pop(stack)
  end
  puts out.join(' ')
end
